hazelcast/src/main/java/com/hazelcast/spi/merge/SplitBrainMergeTypes.java

```diff
@@ -49,18 +49,22 @@
      *
      * @since 3.10
      */
-    public interface MapMergeTypes extends MergingEntry<Data, Data>, MergingCreationTime<Data>, MergingHits<Data>,
-            MergingLastAccessTime<Data>, MergingLastUpdateTime<Data>, MergingTTL<Data>, MergingMaxIdle<Data>, MergingCosts<Data>,
-            MergingVersion<Data>, MergingExpirationTime<Data>, MergingLastStoredTime<Data> {
+    public interface MapMergeTypes extends
+            MergingEntry<Object, Object>,
+            MergingCreationTime<Object>, MergingHits<Object>,
+            MergingLastAccessTime<Object>, MergingLastUpdateTime<Object>,
+            MergingTTL<Object>, MergingMaxIdle<Object>, MergingCosts<Object>,
+            MergingVersion<Object>, MergingExpirationTime<Object>, MergingLastStoredTime<Object> {
```

Comment:
Putting here what makes sense to me as an another alternative:

1. +1 for not extending `MergingValue`
2. Use user friendly generic type for all interfaces in `SplitBrainMergeTypes.java` helper class. And change `MergingValue` like below:
```
public interface MapMergeTypes<K, V> extends MergingEntry<K, V>, MergingCreationTime<V>, MergingHits<V>,
            MergingLastAccessTime<V>, MergingLastUpdateTime<V>, MergingTTL<V>, MergingMaxIdle<V>, MergingCosts<V>,
            MergingVersion<V>, MergingExpirationTime<V>, MergingLastStoredTime<V> {
    }
```


```
public interface MergingValue<V> {

// return value with user friendly type. User friendly type of value is the type you give when creating IMap<K,V>
    V getValue();

   // return raw value, raw value type depends on the memory format of  //backing storage, this can be used for performance reasons.
    Object getRawValue();
}
```

Reply:
Right, good comments, I'll try to make it so. So I guess the combination of the two would actually make it look like:
```
    public interface MapMergeTypes<K, V> extends MergingEntry<K, V>,
            MergingCreationTime, MergingHits,
            MergingLastAccessTime, MergingLastUpdateTime,
            MergingTTL, MergingMaxIdle, MergingCosts,
            MergingVersion, MergingExpirationTime,
            MergingLastStoredTime {
    }
```
Agree @ahmetmircik @vojtechtoman ?